yokota/ml_mnist/maximun_likely_estimator.py

```diff
@@ -0,0 +1,55 @@
+#!/user/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+from sklearn.utils import shuffle
+from sklearn.metrics import f1_score
+from sklearn.datasets import fetch_mldata
+from sklearn.cross_validation import train_test_split
+
+import numpy as np
+
+
+mnist = fetch_mldata('MNIST original')
+mnist_X, mnist_y = shuffle(mnist.data, mnist.target.astype('int32'))
+
+mnist_X = mnist_X / 255.0
+
+train_X, test_X, train_y, test_y = train_test_split(mnist_X, mnist_y, test_size=0.2, random_state=44)
+ids_x = np.array([train_y == i for i in range(10)])
+means = None
+variances = None
+inv_vars = None
+for ids in ids_x:
+  x = train_X[ids]
+  mean_x = np.sum(x, axis=0) / len(x)
+  _x = np.array([f - mean_x for f in x])
+  variance_x = np.array([np.dot(_x.T, _x) / len(_x)])
+  if means == None:
```

Comment:
maximun_likely_estimator.py:28: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if means == None: